### ISSUE TITLE
`go_binary()` with a cgo dependency fails to link when go_prefix is ""

### ISSUE THREAD

**Issue by treaster** [bazelbuild/rules_go#254]
`go_binary()` with a cgo dependency fails to link when go_prefix is "".

The error message is `"gcc: error: missing argument to '-L'.`

This can be traced to https://github.com/bazelbuild/rules_go/blob/master/go/def.bzl#L383 If prefix is empty, the `-L` flag doesn't get a proper value.

`go build` is able to build the source code correctly.
bazel is able to build go_binary() targets that don't have a cgo dependency.

```
$ bazel version
Build label: 0.4.3- (@non-git)
Build target: bazel-out/local-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Jan 30 22:07:29 2017 (1485814049)
Build timestamp: 1485814049
Build timestamp as int: 1485814049
```

Instructions to reproduce the problem can be found at https://github.com/treaster/bazel_cgo_example, also copied below:

1. `git clone https://github.com/treaster/bazel_cgo_example.git && export GOPATH=$(pwd)/bazel_cgo_example && cd $GOPATH/src`
2. `go build example.com/cgo_bad_app # Building with go build works`
3. `bazel build example.com/go_ok_app/... # bazel configuration works`
4. `bazel build example.com/cgo_ok_lib/... # cgo can compile the library target`
5. `bazel build example.com/cgo_bad_app/... # cgo fails to link the application`

**Comment by pmbethe09** [comment]
Is this only when go_prefix is ""?  In that case I don't see much value in fixing this -- when would you ever have an empty prefix?

**Comment by treaster** [comment]
Yes, only when go_prefix is "".

I admit that I don't fully understand the go_prefix model, and I may be doing it not-quite-right*. However, it's a little weird to me that my configuration works with pure Go code, but it fails to work with cgo. This makes it feel like a bazel bug rather than a configuration issue.

I *think* it's an easy fix - just check that `prefix != ""` before adding the `-L` flag at def.bzl:383. But I don't feel l understand the rules_go code well enough to say that assertively. I've made the change locally and it solves my problem, but I can't be sure that I'm not breaking something else.

*I'm happy to have a discussion of why I'm using an empty go_prefix. I expect I'd learn a lot. But I think it's outside the scope of this specific issue.

**Comment by pmbethe09** [comment]
go_prefix essentially aligns with the root of the version-control directory.
e.g. github.com/bazelbuild/buildifier.git is a git repo, so BUILD files start at that directory and use
go_prefix("github.com/bazelbuild/buildifier")

So in your example, it would be more likely to put the WORKSPACE and root BUILD file in the example.com directory as opposed to above it -- and then go_prefix("example.com")

The point being, I am really not worried about bugs with empty go_prefix as I don't know of a single real example where you would want to use that.

**Comment by treaster** [comment]
Thanks again for quick response!

I've opened a thread on bazel-discuss to discuss the validity of my approach, but if you feel it's cleaner to have the discussion in the context of this issue report, that's totally okay with me. Just let me know which you prefer.

https://groups.google.com/forum/#!topic/bazel-discuss/Iug4iVofCps

**Comment by jayconrod** [comment]
I think this was fixed by #277 but please re-open if it's still an issue.
